app/Dockerfile

```diff
@@ -1,31 +1,65 @@
-# Use the Active LTS version of Node.
-# See https://nodejs.org/en/about/releases/
-FROM node:16-alpine
-# Keep container packages up-to-date.
-# -U runs both apk update and apk upgrade.
-RUN apk -U upgrade
-
-
-# setup user stuff
-ARG RUN_UID
-ARG RUN_USER
-
-# The following logic creates the RUN_USER home directory and the directory where
-# we will be storing the application in the image. This runs when the user is not root
-RUN : "${RUN_USER:?RUN_USER and RUN_UID need to be set and non-empty.}" && \
-  [ "${RUN_USER}" = "root" ] || \
-  (addgroup -S "${RUN_USER}" && adduser -S -G "${RUN_USER}" -h "/home/${RUN_USER}" -u ${RUN_UID} "${RUN_USER}" \
-  && mkdir /app \
-  && chown -R "${RUN_USER}:${RUN_USER}" "/home/${RUN_USER}" /app)
-
-USER ${RUN_USER}
+# Dockerfile for building a production image for deployment.
+# This file is largely based on the Next.js Docker example:
+# https://github.com/vercel/next.js/blob/canary/examples/with-docker-compose
+# =============================================================================
+FROM node:18-alpine AS base
+
+# =============================================================================
+# Stage 1: Build the Next.js app
+# =============================================================================
+FROM base AS builder
```

Comment:
Why not have the dev version in the same file, but you specify the target in the compose files?

Reply:
No reason beyond lack of Docker familiarity and this is how the Next.js example was in their repo 😄